src/lib/node.ts

```diff
@@ -54,83 +58,152 @@ export let NODE_TYPE = {
 	URL: "url",
 } as const
 
+/**
+ * a type of NodeType which gets its values from the object of node types.
+ * this lets us write functions that take a `NodeType` as an argument.
+ */
 export type NodeType = typeof NODE_TYPE[keyof typeof NODE_TYPE]
 
+/**
+ * The base node from which other nodes extend.
+ * abstract.
+ */
 export interface Node {
+	/** a type from the set of valid node types */
 	type: NodeType
+	/**
+	 * a source object as defined in `get.ts`.
+
+	 * these provide the file, json path, line and column where a piece of
+	 * information orignated
+	 */
 	source: Source
+	/**
+	 * a list of failed expectations that are not fatal, but for which we
+	 * might provide some helpful insights
+	 */
 	opinions?: Opinion[]
 }
 
-export interface Value<T> extends Node {
-	type: NodeType
-	value: T
+/** a node that represents a raw value.
+ *
```

Comment:
I'm not sure if it matters syntax-wise but just in case:
```suggestion
/**
 * a node that represents a raw value.
```